src/reducers/fileTree.tsx

```diff
@@ -372,6 +376,75 @@ export const goToRelativeFile = ({
   };
 };
 
+type GoToRelativeMessageParams = {
+  _getRelativeMessage?: typeof getRelativeMessage;
+  _viewVersionFile?: typeof viewVersionFile;
+  currentMessageUid: LinterMessage['uid'] | void;
+  currentPath: string;
+  messageMap: LinterMessageMap;
+  pathList: string[];
+  position: RelativePathPosition;
+  versionId: number;
+};
+
+export const goToRelativeMessage = ({
+  /* istanbul ignore next */
+  _getRelativeMessage = getRelativeMessage,
+  /* istanbul ignore next */
+  _viewVersionFile = viewVersionFile,
+  currentMessageUid,
+  currentPath,
+  messageMap,
+  pathList,
+  position,
+  versionId,
+}: GoToRelativeMessageParams): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const nextRelativeMessageInfo = _getRelativeMessage({
+      currentMessageUid,
+      currentPath,
+      messageMap,
+      pathList,
+      position,
+    });
+
+    if (nextRelativeMessageInfo) {
+      const { line, path, uid } = nextRelativeMessageInfo;
+      const {
+        router: { location },
+      } = getState();
+
+      const queryParams = queryString.parse(location.search);
+      const newParams = { ...queryParams };
+      newParams.path = path;
```

Comment:
It would be helpful to create a constant for `'path'` and share it with this code:

https://github.com/mozilla/addons-code-manager/blob/d821eeb670e7a24b321654684a4f40fe802ff67a/src/utils.tsx#L56

Otherwise, it's hard to grep for `path` and find all the times it's used as a query string parameter. While you're at it, I'd suggest making `messageUid` a constant, too.

Reply:
I _think_ I understand what you're asking, so I've implemented it. You can let me know if I got it right when I push the next commit.